### PR TITLE
Code formatting with "black"

### PR DESCRIPTION
**Pull request by Fernerkundung**
This PR is an example for code formatting with `black` related to #295 and can be regarded as a proposal for a reference code formatting implementation.

- formatted all code with `black`
- included Travis check for code style
- document the usage of `black`

Please consider this as a basis for further discussion @valgur @j08lue.

### CODE REVIEW

**Comment by Fernerkundung** [comment]
Issues that appeared:
- black requires Python >= 3.6, this would probably better be solved with a built matrix than bash if/else
- black adds trailing commas when after keyword args when the build target is >Py34, therefore target needs to be the lowed Python3 version we support

File: .travis.yml [hunk 1]
```diff
@@ -18,8 +18,12 @@ install:
 - pip install codecov
 # skip Pandas on Python <3.4
 - if [[ $TRAVIS_PYTHON_VERSION == 3.4 ]]; then pip install -e .[dev34]; else pip install -e .[dev]; fi
+# install black on 3.6
```
Comment (j08lue) [inline .travis.yml]: u mean 3.7

File: docs/conf.py [hunk 2]
```diff
@@ -20,227 +20,225 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath('..'))
+sys.path.insert(0, os.path.abspath(".."))
```
Comment (j08lue) [inline docs/conf.py]: arrg, I do prefer single quotes, but hey...

Comment (j08lue) [reply docs/conf.py]: There *is* an option for leaving quotes alone https://github.com/psf/black/issues/118#issuecomment-393298377, but that means that they are not auto-formatted when someone uses them differently.

File: .travis.yml (hunk 1 above)
Comment (Fernerkundung) [reply .travis.yml]: yes, I do.

**Comment by j08lue** [comment]
Can we quickly fix this up and merge, @Fernerkundung and @valgur? Might come in handy here in Hacktober... I can live with double quotes - the benefits of automatic formatting by far outweigh the drawbacks on this project with various contributors of varying experience...

**Comment by valgur** [comment]
I personally prefer single quotes as well, but it's not a dealbreaker for me. I support merging this.

File: tests/test_download.py [hunk 3]
```diff
@@ -15,47 +15,46 @@
...
 def test_trigger_lta_success(http_status_code):
     api = SentinelAPI("mock_user", "mock_password")
     request_url = "https://scihub.copernicus.eu/apihub/odata/v1/Products('8df46c9e-a20c-43db-a19a-4240c2ed3b8b')/$value"
 
     with requests_mock.mock() as rqst:
-        rqst.get(
-            request_url,
-            status_code=http_status_code
-        )
+        rqst.get(request_url, status_code=202)
```
Comment (j08lue) [inline tests/test_download.py]: oops, made a mistake during merge conflict resolution here. Will fix shortly.

**Comment by j08lue** [comment]
@Fernerkundung, will you pull the trigger?

**Comment by j08lue** [comment]
Ah, I guess you approved with your +1 on @valgur's approval. Merging now then.

**Pull request merged**